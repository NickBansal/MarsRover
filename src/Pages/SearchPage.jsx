import React, { Component } from 'react'
import Navbar from '../Components/Navbar'
import * as api from '../api'

class SearchPage extends Component {

    state = {
        searchTerm: "",
        searchItems: []
    }

    render() {
        const { searchTerm } = this.state
        const pageTitle = !searchTerm ? 'Please begin your search' : searchTerm
        return (
            <div>
                <Navbar handleSubmit={this.handleSubmit} />
                <h1>{pageTitle}</h1>
            </div>
        )
    }

    handleSubmit = (searchTerm) => {
        this.setState({
            searchTerm
        })
    }

    componentDidUpdate(prevProps, prevState) {
        if (prevState.searchTerm !== this.state.searchTerm) {
            api.getPictures(this.state.searchTerm)
            .then(items => {
                this.setState({
                    searchItems: items.collection.items
                })
            })
            .catch(error => {
                console.log(error, "<<<< ERROR")
            })
        }
    }
}

export default SearchPage