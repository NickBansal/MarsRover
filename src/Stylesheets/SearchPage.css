.SingleThumbnail {
    width: 200px;
    height: 261px;
    display: inline-block;
    margin: 10px;
    border: 1px solid white;
    border-top-left-radius: 9px;
    border-top-right-radius: 9px;
    vertical-align: top;
}

.SingleThumbnailTitle {
    width: 100%;
    height: 50px;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
        -ms-flex-pack: center;
            justify-content: center;
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
}

.SingleThumbnail:hover {
    background: lightgrey;
    color: #162e4a;
    cursor: pointer;
    -webkit-filter: grayscale(90%);
            filter: grayscale(90%);
    -webkit-transition: 0.5s;
    -o-transition: 0.5s;
    transition: 0.5s
}

.SingleThumbnail img {
    height: 200px;
    width: 100%;
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
    border: 5px white solid
}

.SingleThumbnail p {
    font-size: 15px;
    margin: 0;
    padding: 0 10px;
}

.AllSearchItems {
    padding: 0 50px;
    height: 80vh;
    overflow: scroll;
}

.Results {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: justify;
        -ms-flex-pack: justify;
            justify-content: space-between;
    font-size: 22px;
    color: white;
    padding: 7px 20px;
}

.Results p {
    margin: 0
}

a {
    color: white
}

.EnterSearch, .ErrorMessage, .EmptyMessage {
    opacity: 0;
    -webkit-animation: 1 fadein 1.5s forwards;
            animation: 1 fadein 1.5s forwards;
    font-size: 25px
}

.RadioSelect {
    margin-right: 20px;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
}

.RadioSelect input {
    margin-right: 5px;
    font-size: 1rem;
}

@-webkit-keyframes fadein {
    100% {
        opacity: 1;
    }
}

@keyframes fadein {
    100% {
        opacity: 1;
    }
}

@media (max-width: 700px) {
    .SingleThumbnail {
        width: 170px;
        height: 220px;
    }
    .SingleThumbnail img {
        height: 170px;
        border: 2px white solid
    }
    .SingleThumbnail p {
        font-size: 13px;
    }

    .SingleThumbnailTitle {
        height: 40px; 
    }

    .Results {
        font-size: 17px;
        -webkit-box-orient: vertical;
        -webkit-box-direction: reverse;
            -ms-flex-direction: column-reverse;
                flex-direction: column-reverse;
        -webkit-box-pack: center;
            -ms-flex-pack: center;
                justify-content: center;
        -webkit-box-align: center;
            -ms-flex-align: center;
                align-items: center
    }
    .Results p {
        margin-bottom: 5px;
    }
}

@media (max-width: 600px) {
    p.EnterSearch, p.ErrorMessage, p.EmptyMessage {
        font-size: 20px
    }
    .AllSearchItems {
        height: 75vh
    }
}

@media (max-width: 500px) {
    .SingleThumbnail {
        display: -webkit-box;
        display: -ms-flexbox;
        display: flex;
        width: 100%;
        height: 120px;
        border-radius: 5px;
        -webkit-box-align: center;
            -ms-flex-align: center;
                align-items: center;
        text-align: center;
    }
    
    .SingleThumbnail p {
        font-size: 16px;
        text-align: center; 
        overflow-wrap: break-word;
    }

    .SingleThumbnail img {
        width: 50%;
        height: 120px;
        border-top-left-radius: 3px;
        border-bottom-left-radius: 3px;
        border-top-right-radius: 0;
        border-bottom-right-radius: 0
    }
    a {
        text-decoration: none
    }
    .SingleThumbnailTitle {
        height: 100%; 
    }
}