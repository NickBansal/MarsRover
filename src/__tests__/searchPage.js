import '../setupTest';
import React from 'react'
import SearchPage from '../Pages/SearchPage';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";

describe.only('<SearchPage />', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallow(<SearchPage />)
    })
    it('Component matches the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('Changes state on click handler', () => {
        wrapper.instance().handleClick('video')
        expect(wrapper.state().input).toBe('video')
        wrapper.instance().handleClick('image')
        expect(wrapper.state().input).toBe('image')
        wrapper.instance().handleClick('audio')
        expect(wrapper.state().input).toBe('audio')
    })
    it('Changes state on handle submit', () => {
        wrapper.instance().handleSubmit('Earth')
        expect(wrapper.state().loading).toBe(true)
        expect(wrapper.state().start).toBe(false)
        expect(wrapper.state().allItems.length).toBe(0)
        expect(wrapper.state().searchTerm).toBe('earth')
        expect(wrapper.state().upper).toBe(false)

        wrapper.instance().handleSubmit('Earth')
        expect(wrapper.state().searchTerm).toBe('EARTH')
        expect(wrapper.state().upper).toBe(true)
    })
})