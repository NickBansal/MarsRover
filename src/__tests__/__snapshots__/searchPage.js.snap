// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchPage /> Component matches the snapshot 1`] = `
<div
  className="SearchPage"
>
  <Navbar
    handleSubmit={[Function]}
  />
  <div
    className="AllSearchItems"
  >
    <Router>
      <EnterSearch
        path="/"
      />
      <SingleItem
        path="/:id"
      />
    </Router>
  </div>
</div>
`;
